### PR TITLE
Capture default UserAgent crashes

### PR DESCRIPTION
**Pull request by cmonfortep**
Task/Issue URL: https://app.asana.com/0/488551667048375/1201519018374192/f

### Description
This PR intends to capture a crash when getting WebView default user agent on some devices.

### Steps to test this PR

_No changes_
1. run the app
2. visit any website
3. Filter logs by `User Agent is`
4. Ensure your UA matches the one on your device (e.g: check if chrome version is the one installed on your device)

_Capture crash_
1. Go to `provideDefaultUserAgent` method in `UserAgentProvider`
2. hardcode `throw RuntimeException("")`  before `WebSettings.getDefaultUserAgent(context)`
3. run the app
4. visit any website
5. Ensure app does not crash

### CODE REVIEW

File: app/src/main/java/com/duckduckgo/app/browser/useragent/UserAgentProvider.kt [hunk 1]
```diff
@@ -111,6 +127,7 @@ class UserAgentProvider constructor(@Named("defaultUserAgent") private val defau
         const val SPACE = " "
         val mobilePrefix = "Mozilla/5.0 (Linux; Android ${Build.VERSION.RELEASE})"
         val desktopPrefix = "Mozilla/5.0 (X11; Linux ${System.getProperty("os.arch")})"
+        val fallbackDefaultUA = "$mobilePrefix AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/96.0.4664.104 Mobile Safari/537.36"
```
Comment (cmonfortep) [inline app/src/main/java/com/duckduckgo/app/browser/useragent/UserAgentProvider.kt:130]: I'm using the latest chrome version here as part of our UA. should not be an issue. If anyone knows a better way to handle this, please share.

File: app/src/main/java/com/duckduckgo/app/browser/useragent/UserAgentProvider.kt [hunk 2]
```diff
@@ -96,10 +102,20 @@ class UserAgentProvider constructor(@Named("defaultUserAgent") private val defau
...
+    private fun defaultUserAgent(): String = runCatching {
+        defaultUserAgent.get()
+    }.let { result ->
+        if (result.isFailure) {
+            pixel.fire(AppPixelName.WEBVIEW_DEFAULT_USER_AGENT_CRASH)
+            return fallbackDefaultUA
+        }
+        return@let result.getOrDefault(fallbackDefaultUA)
+    }
+
```
Comment (aitorvs) [inline app/src/main/java/com/duckduckgo/app/browser/useragent/UserAgentProvider.kt:118]: I think this should rather be in the `BrowserModule`. There is no need to make changes to this calss.

```kotlin
    @Provides
    @SingleInstanceIn(AppScope::class)
    fun userAgentProvider(@Named("defaultUserAgent") defaultUserAgent: Provider<String>, deviceInfo: DeviceInfo): UserAgentProvider {
        try {
            return UserAgentProvider(defaultUserAgent, deviceInfo)
        } catch (e: Exception) {
            return ""
        }
    }
```

**Comment by malmstein** [comment]
@cmonfortep besides the fact that it's broken,  can you make sure you fill in the PR template? All the relevant info is missing

**Comment by cmonfortep** [comment]
Thanks, @malmstein, this is still in draft. I will update the description and fix the build before is ready for review.

File: app/src/main/java/com/duckduckgo/app/browser/useragent/UserAgentProvider.kt (hunk 2 above)
Comment (marcosholgado) [reply app/src/main/java/com/duckduckgo/app/browser/useragent/UserAgentProvider.kt:118]: Did you mean change the `DefaultUserAgentModule`? :P

```
@ContributesTo(AppScope::class)
@Module
class DefaultUserAgentModule {
    @SingleInstanceIn(AppScope::class)
    @Provides
    @Named("defaultUserAgent")
    fun provideDefaultUserAgent(context: Context): String {
       try {
           return WebSettings.getDefaultUserAgent(context)
        } {catch (e: Exception) {
            return ""
        }
    }
}

Comment (marcosholgado) [reply app/src/main/java/com/duckduckgo/app/browser/useragent/UserAgentProvider.kt:118]: Yup 👍  to this

Comment (cmonfortep) [reply app/src/main/java/com/duckduckgo/app/browser/useragent/UserAgentProvider.kt:118]: I'm ok not changing the class, however, I don't think is a good idea to provide an empty string as a fallback.

WDTY about wrapping `WebSettings.getDefaultUserAgent(context)` into in our own class that will provide a default UA?

Comment (marcosholgado) [reply app/src/main/java/com/duckduckgo/app/browser/useragent/UserAgentProvider.kt:118]: I think the empty string was an example, we should return the actual UA there.

Comment (cmonfortep) [reply app/src/main/java/com/duckduckgo/app/browser/useragent/UserAgentProvider.kt:118]: I've discussed this with @aitorvs. If we don't need to send a pixel, we can follow this approach.

**Review by marcosholgado — APPROVED** [review]
Nice job! 🚢  🇮🇹

**Pull request merged**